src/integration/java/com/smartcar/sdk/VehicleTest.java

```diff
@@ -0,0 +1,83 @@
+package com.smartcar.sdk;
+
+import com.smartcar.sdk.data.SmartcarResponse;
+import com.smartcar.sdk.data.VehicleInfo;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+/**
+ * Integration Test Suite: /vehicles/:id
+ */
+public class VehicleTest extends IntegrationTest {
+    private Vehicle vehicle;
+
+    @BeforeSuite
+    public void beforeSuite() throws Exception {
+        String[] vehicleIds = AuthClient.getVehicleIds(this.getAuth().getAccessToken()).getData().getVehicleIds();
+        this.vehicle = new Vehicle(vehicleIds[0], this.getAuth().getAccessToken());
+    }
+
+    /**
+     * Tests that vehicle info can be obtained.
+     */
+    @Test
+    public void testInfo() throws SmartcarException {
+        VehicleInfo info = this.vehicle.info();
+    }
+
+    /**
+     * Tests that the vehicle VIN can be obtained.
+     */
+    @Test
+    public void testVin() throws SmartcarException {
+        String vin = this.vehicle.vin();
+    }
+
+    /**
+     * Tests that the application permissions can be obtained.
+     */
+    @Test
+    public void testPermissions() throws SmartcarException {
+        String[] permissions = this.vehicle.permissions();
+    }
+
+    /**
+     * Tests that the odometer value can be obtained.
+     */
+    @Test
+    public void testOdometer() throws SmartcarException {
+        SmartcarResponse response = this.vehicle.odometer();
+    }
+
+    /**
+     * Tests that the vehicle's location can be obtained.
+     */
+    @Test
+    public void testLocation() throws SmartcarException {
+        SmartcarResponse response = this.vehicle.location();
```

Comment:
Can we verify the responses at all? I guess Java takes care of the type checking...

Reply:
It does feel weird to have no assertions. When I wrote them out, I found they weren't really accomplishing anything given that all we care about is that the call works and returns data that can be parsed without errors. I found I was essentially writing API tests at that point, instead of verifying the SDK.

As you said, these tests are leaning on Java's type checking to ensure integrity. With Mock returning random values for most signals, I'm not sure there's much else we can assert.